### DISCUSSION TITLE
Why is the download command always throwing HTTP 401 errors?

### DISCUSSION THREAD

**Discussion by brianwilcken** [dashingsoft/pyarmor#1061] [Q&A]
1. A clear and concise description of the question (required)

Why am I getting an HTTP 401 Unauthorized error message when I run the pyarmor download command?  Do I need to somehow first input account credentials into pyarmor before attempting to download the index.json file from the pyarmor servers?

2. The full pyarmor command and full output log (required if something is wrong)

pyarmor download --help-platform
INFO     PyArmor Version 7.6.1
INFO     Python 3.8.15
INFO     Getting remote file: https://pyarmor.dashingsoft.com/files/r51.5/index.json
ERROR    HTTP Error 401: Unauthorized

3. If distributing the obfuscated script to other machine, which files are copied (optional)

Not Applicable

5. The command to run the obfuscated scripts and full traceback when something is wrong

Not Applicable

**Comment by jondy** [comment]
What's your license code?

If license code < 4000, it's old license and should no problem.
If license code > 4000, it's new license and can't not be used for Pyarmor < 8.0

Please check pinned issue #980

**Reply by brianwilcken** [reply]
My license code is 4025.  We registered our copy of pyarmor last week not realizing that the latest version of pyarmor does not yet support cross-platform obfuscation.  We need to prepare a linux-compatible build but we are running on Windows.  Can we fall back to use pyarmor version 7 for the sake of preparing a linux-compatible build?  So far, every time I try to run cross-platform obfuscation using pyarmor 7.6.1 I am seeing the HTTP 401 error message.  Am I understanding correctly that the reason the 401 error message is appearing is because the license we purchased does not support version 7?  If that is the case then can we purchase a license for version 7 instead?

**Reply by jondy** [reply]
The license can't be downgrade.

Pyarmor 8.1 will be released around April 10, it supports cross platforms.

**Reply by brianwilcken** [reply]
Can we purchase a license for version 7 just to get past this obstacle?  We will also retain our version 8 license going forward but we have a time-sensitive need to prepare a cross-platform build.  We are fine purchasing a version 7 license for an additional fee.

**Reply by jondy** [reply]
Here it's purchase link for old license
https://order.shareit.com/cart/add?vendorid=200089125&PRODUCT[300871197]=1

Use this license with command `pyarmor-7`

**Reply by brianwilcken** [reply]
Thank you
